### ISSUE TITLE
Pathological case of inserting rays in an arrangement

### ISSUE THREAD

**Issue by AgarwalSaurav** [CGAL/cgal#7311]
## Issue Details

I am inserting rays into an arrangement. I get the following runtime behavior when I insert the entire ray vector in a single CGAL::insert function call. However, it works fine if I insert the rays individually. This only happens for the specific rays given below.

> 	terminate called after throwing an instance of 'CGAL::Assertion_exception'
	  what():  CGAL ERROR: assertion violation!
	Expr: v != m_invalid_vertex
	File: CGAL/Surface_sweep_2/Arr_construction_ss_visitor.h
	Line: 987

## Source Code

```cpp
#include <CGAL/Arr_linear_traits_2.h>
#include <CGAL/Arrangement_2.h>
#include <CGAL/Exact_predicates_exact_constructions_kernel.h>

typedef CGAL::Exact_predicates_exact_constructions_kernel K;
typedef K::Point_2 Point_2;
typedef K::Direction_2 Direction_2;
typedef K::Ray_2 Ray_2;
typedef K::Line_2 Line_2;

typedef CGAL::Arr_linear_traits_2<K> Traits_2;
typedef CGAL::Arrangement_2<Traits_2> Arrangement_2;

int main() {

	std::vector<Ray_2> rays;
	rays.push_back(Ray_2(Point_2(180.055441307383,924.254116005402), Direction_2(-229.144941997007,34.8855739060313)));
	rays.push_back(Ray_2(Point_2(156.104233328527,993.005647375856), Direction_2(-0,150.383759942161)));
	rays.push_back(Ray_2(Point_2(156.104233328527,993.005647375856), Direction_2(-60.1375275914886,22.2399807225157)));

	Arrangement_2 arr;

	// The following line causes a segfault

	/* terminate called after throwing an instance of 'CGAL::Assertion_exception' */
	/*   what():  CGAL ERROR: assertion violation! */
	/* Expr: v != m_invalid_vertex */
	/* File: CGAL/Surface_sweep_2/Arr_construction_ss_visitor.h */
	/* Line: 987 */
	CGAL::insert(arr, rays.begin(), rays.end());

	// Inserting individual rays works fine
	/* for (auto const &ray:rays) { */
	/* 	CGAL::insert(arr, ray); */
	/* } */

	return 0;
}
```

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Linux 64
* Compiler: g++ 12.2.1
* Release or debug mode: Debug
* Specific flags used (if any): -g -std=c++17 
* CGAL version: 5.5.2
* Boost version: 1.81.0-3

**Comment by efifogel** [comment]
This has been fixed in PR #7323 <https://github.com/CGAL/cgal/pull/7323>
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by lrineau** [comment]
Fixed by the merge of https://github.com/CGAL/cgal/pull/7323
